domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/TabShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,103 @@
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Component
+public class TabShowConditionValidatorImpl implements DisplayGroupValidator {
+
+    private final ShowConditionParser showConditionParser;
+
+    @Autowired
+    public TabShowConditionValidatorImpl(ShowConditionParser showConditionParser) {
+        this.showConditionParser = showConditionParser;
+    }
+
+    @Override
+    public ValidationResult validate(DisplayGroupEntity thisDisplayGroup, Collection<DisplayGroupEntity> allDisplayGroups) {
+        ValidationResult validationResult = new ValidationResult();
+        List allTabDisplayGroups = getAllTabDisplayGroups(thisDisplayGroup, allDisplayGroups);
+
+        if (tabPreconditions(thisDisplayGroup)) {
+            ShowCondition showCondition;
+            try {
+                showCondition = showConditionParser.parseShowCondition(thisDisplayGroup.getShowCondition());
+            } catch (InvalidShowConditionException e) {
+                validationResult.addError(new DisplayGroupInvalidShowConditionFieldForTab(thisDisplayGroup));
+                return validationResult;
+            }
+
+            String showConditionField = showCondition.getField();
+
+            if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                validationResult.addError(new DisplayGroupInvalidShowConditionFieldForTab(showConditionField, thisDisplayGroup));
+            }
+        }
+
+        if (tabFieldsPreconditions(thisDisplayGroup)) {
+            for (DisplayGroupCaseFieldEntity caseField : thisDisplayGroup.getDisplayGroupCaseFields()) {
+                if (caseField.getShowCondition() != null) {
+                    ShowCondition showCondition;
+                    try {
+                        showCondition = showConditionParser.parseShowCondition(caseField.getShowCondition());
+                    } catch (InvalidShowConditionException e) {
+                        validationResult.addError(new DisplayGroupInvalidShowConditionFieldForTabField(caseField));
+                        return validationResult;
+                    }
+
+                    String showConditionField = showCondition.getField();
+
+                    if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                        validationResult.addError(new DisplayGroupInvalidShowConditionFieldForTabField(showConditionField, caseField));
+                    }
+                }
+            }
+        }
+        return validationResult;
+    }
+
+    private List getAllTabDisplayGroups(DisplayGroupEntity thisDisplayGroup, Collection<DisplayGroupEntity> allDisplayGroups) {
```

Comment:
this seems to me a method that can be useful for others DisplayGroupValidator implementations. Can we move it to the DisplayGroupValidator interface as a default method?

Reply:
I see lots of methods to deal with display groups here. Another approach, perhaps better than the default method on the DisplayGroupValidator, is to introduce a 'utility' class DisplayGroups, and move all these methods that operate on display groups there

Reply:
Hmm I do not see them being used outside of this context as they are quiet customised. I agree we should always strive for extracting such methods but because of the above and due to time constraints I will leave it as is if it's ok with you?